worker.sml

```diff
@@ -314,7 +314,7 @@ in
                           handle e as OS.SysErr _ => (API.post (Append(id, exnMessage e)); false)
           in
             if entered andalso system_capture holmake_cmd then
-              (map ((upload id) o trimr) artefacts;
+              (if is_master then map ((upload id) o trimr) artefacts else [];
```

Comment:
`if is_master then app ((upload id) o trimr) artefacts else ()` would be clearer

Reply:
Ah, I wasn't aware of `app`. Its been updated accordingly.